File: euclid1.c
/*  Algorytm Euklidesa. */

#include <stdio.h>

int main()
{
   int a, b;

   printf("Podaj dwie liczby calkowite: ");
   scanf("%d %d", &a, &b);

   while (a != b)
      if (a > b) a = a - b;
      else       b = b - a;

   printf("NWD = %d\n", a);

   return 0;
}

File: euclid1.f
      PROGRAM EUCLID1
c     Algorytm Euklidesa w jezyku Fortran 77
   
      WRITE (*,*) 'Podaj dwie liczby calkowite: '
      READ (*,*) N, M
   
      DO WHILE ( N .NE. M ) 
         IF ( N .GT. M ) THEN 
            N = N - M
         ELSE
            M = M - N
         ENDIF
      ENDDO
      WRITE (*,*) 'NWD=', N
      END
